mava/components/jax/building/system_init.py

```diff
@@ -0,0 +1,272 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+from dataclasses import dataclass, field
+from typing import Any, Callable, List, Optional, Union
+
+from mava.components.jax import Component
+from mava.core_jax import SystemBuilder
+from mava.utils import enums
+from mava.utils.sort_utils import sample_new_agent_keys, sort_str_num
+
+
+class BaseSystemInit(Component):
+    @abc.abstractmethod
+    def __init__(
+        self,
+        config: Any,
+    ):
+        """Initialise system init components.
+
+        Args:
+            config : a dataclass specifying the component parameters.
+        """
+        self.config = config
+
+    @abc.abstractmethod
+    def on_building_init(self, builder: SystemBuilder) -> None:
+        """Compute and add network sampling information to the builder."""
+        pass
+
+    @staticmethod
+    def name() -> str:
+        """_summary_"""
+        return "system_init"
+
+
+@dataclass
+class FixedNetworkSystemInitConfig:
+    shared_weights: bool = True
+    network_sampling_setup: enums.NetworkSampler = (
+        enums.NetworkSampler.fixed_agent_networks
+    )
```

Comment:
Seems like these enums are still necessary for now? I will shortly be working on the dependency guardrails. Up to you whether or not you'd like to remove the enum checks, but I'm happy to leave them in for now.

Reply:
The enums are still needed for now and need to be added to the builder store due to the `DataServer` component needing to access `builder.store.network_sampling_setup_type` [here](https://github.com/instadeepai/Mava/blob/ddcd87db9ece5aa975d51420440dd63b158e8d62/mava/components/jax/building/data_server.py#L54).

Reply:
Is that something I should change in my global configs PR then? If this one gets merged first

Reply:
Yes, I think that would be good.